torch/lib/c10d/FileStore.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <sys/types.h>
+
+#include <map>
+
+#include "Store.hpp"
+
+namespace c10d {
+
+class FileStore : public Store {
+ public:
+  explicit FileStore(const std::string& path);
+
+  virtual ~FileStore();
+
+  void set(
+      const std::string& key,
+      const std::vector<uint8_t>& value) override;
+
+  std::vector<uint8_t> get(const std::string& key) override;
+
+  int64_t add(const std::string& key, int64_t value) override;
+
+  bool check(const std::vector<std::string>& keys) override;
+
+  void wait(
+      const std::vector<std::string>& keys,
+      const std::chrono::milliseconds& timeout = kDefaultTimeout) override;
+
+ protected:
+  std::string path_;
+  off_t pos_;
+
+  std::map<std::string, std::vector<uint8_t>> cache_;
```

Comment:
we might want to use `unordered_map` if we're not using the ordering of keys anywhere. It's generally faster.